joblib/hashing.py

```diff
@@ -146,7 +173,15 @@ def __init__(self, hash_name='md5', coerce_mmap=False):
         else:
             self._getbuffer = memoryview
 
-    def save(self, obj):
+    def persistent_id(self, obj):
+        # Separate dtypes shared by objects in the pickle graph,
+        # to avoid cache invalidation when unpickling a subobject of the graph
+        if isinstance(obj, self.np.dtype):
+            return obj.str
```

Comment:
@ogrisel : returning a pickle.dumps here would be more safe, but slower. What do you think: should we do that, or check stick with the 'str', following a rule of thumb that "what reprs together hashes together"